ext/serializer.c

```diff
@@ -1117,9 +1117,15 @@ static void dd_set_entrypoint_root_span_props_end(zend_array *meta, int status,
 
 static void _serialize_meta(zval *el, ddtrace_span_data *span, zend_string *service_name) {
     bool is_root_span = span->std.ce == ddtrace_ce_root_span_data;
-    zval meta_zv, *meta = &span->property_meta;
+    bool is_inferred_span = span->std.ce == ddtrace_ce_inferred_span_data;
+    zval meta_zv, *meta = &span->property_meta, *inferred_span_zv = NULL;
```

Comment:
```suggestion
    zval meta_zv, *meta = &span->property_meta;
    ddtrace_span_data *inferred_span = NULL;
    if (is_root_span) {
        inferred_span = ddtrace_get_inferred_span(span->root);
    }
```
Let's check it consistently the same way.